src/components/services/Services.tsx

```diff
@@ -55,6 +60,37 @@ export function Services(props: Props) {
     dispatch(checkCompromisedKeys(navigation)).catch(e => {
       console.warn('checkCompromisedKeys error:', e)
     })
+
+    // FIO handle registration:
+    const { freeRegApiToken = '', freeRegRefCode = '' } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
+    const fioWalletsWithAddresses = (
+      await Promise.all(
+        Object.values(account.currencyWallets).reduce<Array<Promise<EdgeCurrencyWallet>>>((promises, wallet) => {
```

Comment:
Never iterate the `account.currencyWallets` object. Always use `account.currencyWalletIds` to drive iteration on this object, and then do `account.currencyWallets[walletId]` to read the actual wallet, which might be `undefined` if it hasn't loaded yet. This has to do with the way the core bridges this particular object.